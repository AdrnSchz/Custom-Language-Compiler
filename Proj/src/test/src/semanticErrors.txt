fact bro const = 0.
fact bro const2.

bro fibonacci ¿bro num? ¡
    sis inode.
    like ¿num >= 0? ¡
    	inode = 3.
    !
    whatever ¡
    	inode += 1.
    !
!

fact sis hello ¿? ¡
   throwback 'a'.
!
CEO ¡
    const = 1.
    sis res = fibonacci¿?.
    
    like ¿ 3 ? ¡
    	test ¿?.
    !
    bro fibonacci = 0´3.
    
!