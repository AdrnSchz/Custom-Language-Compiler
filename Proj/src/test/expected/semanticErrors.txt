------------------------------ What a boomer ¬_¬ ------------------------------

Line 25 | Column 0 -> Missing throwback statement in function: fibonacci

Line 0 | Column 0 -> Function cannot be declared as constant

Line 25 | Column 0 -> Variable: const is a constant, you cannot assign it a value

Line 25 | Column 0 -> Function: fibonacci expects 1 parameters and receives 0

Line 0 | Column 0 -> Condition must be a bipolar expression

Line 25 | Column 0 -> Function: test is not declared

Line 25 | Column 0 -> Variable "fibonacci" expects bro and receives dec_lit

------------------------------ A little sus ◔_◔ ------------------------------

Line 25 | Column 0 -> Variable: const2, is being declared as a constant without being initialized

Line 25 | Column 0 -> Variable: inode, might not be initialized


